### PR TITLE
feat: add listings endpoint with pagination, filters, and tests

### LINKED ISSUE #19: Create Endpoint to List Active Collateral Listings
## **Description:**

Implement a backend API endpoint to **retrieve all active collateral listings** available in the lending protocol. This endpoint will be used by the frontend to display listings that borrowers have made available for lenders to make offers on.

The endpoint should return relevant metadata for each listing and support pagination to handle large datasets efficiently.

## **Endpoint Requirements:**

- **Route:** `GET /listings`
- **Functionality:**
  - Fetch and return all **active** listings.
  - Return key listing fields: `listing_id`, `collateral_contract`, `collateral_id`, `borrower_address`, `loan_amount`, `loan_duration`, `created_at`, and `status`.
  - Support basic **pagination**.
  - Optional query parameters:
    - `collateral_contract` — to filter by NFT contract.
    - `borrower_address` — to filter listings created by a specific user.

## **Acceptance Criteria:**

- The endpoint returns a paginated list of all active listings.
- Only listings with valid and active status are returned.
- Filters work as expected when query parameters are provided.
- Appropriate error handling and status codes for edge cases.
- Covered by unit tests and basic integration test.

### PR DESCRIPTION
**Pull request by daveylupes**
## Description
Implement the `GET /listings` endpoint to fetch all active collateral listings in the lending protocol. This endpoint supports pagination and filtering by `collateral_contract` and `borrower_address`. Returned fields include `listing_id`, `collateral_contract`, `collateral_id`, `borrower_address`, `loan_amount`, `loan_duration`, `created_at`, and `status`.

## Related Issue
Fixes #19  

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Performance improvement
- [ ] Code refactoring
- [ ] Documentation update
- [x] Test coverage improvement
- [ ] CI/CD related changes

## How Has This Been Tested?
- [x] Unit tests
- [ ] Integration tests
- [x] Manual testing

Tests are located in `core/tests/test_listings.py` and include:
- Listing only `active records`.
- Filtering by `collateral_contract`.
- Filtering by `borrower_address`.

Run this command for single test of the function implemented;
```bash
docker compose run --rm trajectfi pytest core/tests/test_listings.py::test_list_active_listings
```

## Python Dependencies

```
# New dependencies:
pytest-django = "^4.11.1"
```

## Starknet-Specific Considerations
N/A

## Checklist
- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

## Security Considerations
- [x] This PR introduces no security concerns
- [ ] This PR modifies authentication/authorization
- [ ] This PR handles sensitive data
- [ ] This PR modifies financial transaction logic

## Performance Impact
- [x] This PR has no significant performance impact
- [ ] This PR improves performance
- [ ] This PR may impact performance negatively (explain why it's necessary)

### CODE REVIEW

File: core/serializers.py [hunk 1]
```diff
@@ -234,3 +235,27 @@ def validate(self, attrs):
...
+
+class ListingSerializer(serializers.ModelSerializer):
+    listing_id = serializers.UUIDField(source="id", read_only=True)
+    collateral_contract = serializers.CharField(
+        source="nft_contract_address", read_only=True
+    )
+    collateral_id = serializers.IntegerField(source="nft_token_id", read_only=True)
+    borrower_address = serializers.CharField(source="user.public_key", read_only=True)
+    loan_amount = serializers.IntegerField(source="borrow_amount", read_only=True)
+    loan_duration = serializers.IntegerField(source="duration", read_only=True)
```
Comment (jaykayudo) [inline core/serializers.py:248]: Please remove these attributes. simply defining them in the Meta class fields attribute is okay.

File: core/serializers.py [hunk 2]
```diff
@@ -234,3 +235,27 @@ def validate(self, attrs):
...
+
+    class Meta:
+        model = Listing
+        fields = [
+            "listing_id",
```
Comment (jaykayudo) [inline core/serializers.py:253]: lisiting id does not exist in the model. use `id` instead.

**Review by jaykayudo — CHANGES_REQUESTED** [review]
@daveylupes Great job so far.
Please attend to the review changes and then this PR is ready to be merged

**Review by jaykayudo — APPROVED** [review]
Look good to me!
Thank you for contibuting to this project.
